### PR TITLE
Refactor molecules to pdf into utility module

### PR DESCRIPTION
**Pull request by SimonBoothroyd**
## Description

This PR refactors the code to create a PDF of a set of molecules from the data set `visualize` method into its own utility. This utility is then use to expose a `visualize` method through the result collection API.

## Status
- [X] Ready to go

### CODE REVIEW

File: openff/qcsubmit/tests/utils/test_visualize.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    _create_openeye_pdf,
+    _create_rdkit_pdf,
+    molecules_to_pdf,
+)
+
+
+@pytest.mark.parametrize(
+    "create_function", [molecules_to_pdf, _create_openeye_pdf, _create_rdkit_pdf]
+)
+def test_create_pdb_function(tmpdir, create_function):
```
Comment (jthorton) [inline openff/qcsubmit/tests/utils/test_visualize.py:16]:
```suggestion
def test_create_pdf_function(tmpdir, create_function):
```

File: openff/qcsubmit/results/results.py [hunk 2]
```diff
@@ -218,6 +219,39 @@ def filter(self, *filters: "ResultFilter"):
...
+
+        Parameters:
+            file_name: The name of the pdf file which will be produced.
+            columns: The number of molecules per row.
+            toolkit: The backend toolkit to use when producing the pdf file.
+        """
+
+        # We filter by inchi key to make sure that we don't double count molecules
+        # with different orderings.
+        unique_smiles = set(
```
Comment (jthorton) [inline openff/qcsubmit/results/results.py:240]: would it be useful to add an API point for the list of unique smiles, I think I have seen this on a few PRs.

**Review by jthorton — APPROVED** [review]
LGTM! Just a minor typo to fix. Great idea putting this on the results object as well!

File: openff/qcsubmit/results/results.py (hunk 2 above)
Comment (SimonBoothroyd) [reply openff/qcsubmit/results/results.py:240]: Good call! I'll handle this in a separate PR after this is merged.

**Pull request merged**